### PR TITLE
Edugonz/api ts sample

### PR DESCRIPTION
**Pull request by edugonz**
Adds API Typescript sample

### CODE REVIEW

File: APISamples/typescript/package.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "pbirs-web-sample",
+  "version": "1.0.0",
+  "description": "PBIRS Web Sample",
```
Comment (rvpatkar) [inline APISamples/typescript/package.json]: why PBIRS?

File: APISamples/typescript/src/api.ts [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        switch (fileInfo.extension.toLocaleLowerCase()) {
+            case 'rdl':
+                item['@odata.type'] = '#Model.Report';
+                break;
+            case 'rsd':
+                item['@odata.type'] = '#Model.DataSet';
+                break;
+            case 'rds':
+                item.ContentType = 'text/xml';
+                item['@odata.type'] = '#Model.Resource';
```
Comment (parthsha) [inline APISamples/typescript/src/api.ts]: Shouldn't this be DataSource?

File: CustomSecuritySample/obj/Debug/CoreCompileInputs.cache [hunk 3]
```diff
@@ -0,0 +1 @@
+1c0d3f52cc59b90e5229f408d9bc7e74fe63e35c
```
Comment (parthsha) [inline CustomSecuritySample/obj/Debug/CoreCompileInputs.cache]: Do we need to check this in?

File: APISamples/typescript/src/api.ts [hunk 4]
```diff
@@ -0,0 +1,170 @@
+import { CatalogItem } from './api';
+import axiosLib, { AxiosInstance } from "axios";
+export class Api {
+    private serverUrl:string;
+    private axios:AxiosInstance;
+
+    constructor(server:string = 'http://localhost/reports'){
```
Comment (rvpatkar) [inline APISamples/typescript/src/api.ts]: should probably go in readme to update as needed

File: APISamples/typescript/src/api.ts (hunk 2 above)
Comment (edugonz) [reply APISamples/typescript/src/api.ts]: DataSources can't be uploaded, they need to be created using the API

File: APISamples/typescript/README.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
+# Getting started
+This project showcases the use of the Report Server API (June 2017 Preview) to build a simple website that lists CatalogItems in the report server, uploads new items, and gets the current user info.
```
Comment (mgmeyer) [inline APISamples/typescript/README.md]: August preview :)

File: APISamples/typescript/src/api.ts [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+                item['@odata.type'] = '#Model.DataSet';
+                break;
+            case 'rds':
+                item.ContentType = 'text/xml';
+                item['@odata.type'] = '#Model.Resource';
+                break;
+            case 'rsc':
+                item['@odata.type'] = '#Model.Component';
+                break;
+            case 'smdl':
```
Comment (mgmeyer) [inline APISamples/typescript/src/api.ts]: You should remove smdl it is not longer supported.

File: APISamples/typescript/src/api.ts [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+export const api = new Api();
+
+
+export enum CatalogItemType { 
+    Unknown = "Unknown",
+    Folder = "Folder",
+    Report = "Report",
+    DataSource = "DataSource",
+    DataSet = "DataSet",
+    Component = "Component",
```
Comment (mgmeyer) [inline APISamples/typescript/src/api.ts]: Is this an actual type, I don't think I've ever seen the Component catalog item type.

File: APISamples/typescript/src/home.tsx [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+            </div>
+        )
+    }
+}
+
+let CatalogItemTile:React.SFC<{item:CatalogItem}> = ({item}) => {
+    let tagColor;
+
+    switch(item.Type){
+        case CatalogItemType.Report:
```
Comment (mgmeyer) [inline APISamples/typescript/src/home.tsx]: Should we add excel to this? Also what exactly is it doing, looks like icon support but seems odd to use info, warning and error as icons.

Comment (edugonz) [reply APISamples/typescript/src/home.tsx]: It just for visual guidance, the 'info','warning' and 'error' tags are just a way of naming the usual 'blue', 'yellow, and 'red' colors

File: APISamples/typescript/src/api.ts (hunk 7 above)
Comment (edugonz) [reply APISamples/typescript/src/api.ts]: It's specified in the API, and the portal also has it defined

**Pull request merged**